server.py

```diff
@@ -82,42 +141,50 @@ def do_get_responses():
         per_page = int(per_page)
 
     search_criteria = {}
-    if survey_id: search_criteria['survey_response.survey_id'] = survey_id
-    if form:      search_criteria['survey_response.form'] = form
-    if ru_ref:    search_criteria['survey_response.metadata.ru_ref'] = ru_ref
-    if period:    search_criteria['survey_response.collection.period'] = period
-    if added_ms:  search_criteria['added_date'] = { "$gte" : datetime.fromtimestamp(int(added_ms)/1000.0) }
+    if survey_id:
+        search_criteria['survey_response.survey_id'] = survey_id
+    if form:
+        search_criteria['survey_response.form'] = form
+    if ru_ref:
+        search_criteria['survey_response.metadata.ru_ref'] = ru_ref
+    if period:
+        search_criteria['survey_response.collection.period'] = period
+    if added_ms:
+        search_criteria['added_date'] = {
+            "$gte": datetime.fromtimestamp(int(added_ms) / 1000.0)
+        }
 
     results = {}
-    responses  = []
-    count = db.responses.find(search_criteria).count()
+    responses = []
+    db_responses = get_db_responses()
+    count = db_responses.find(search_criteria).count()
     results['total_hits'] = count
-    cursor = db.responses.find(search_criteria).skip(per_page*(page-1)).limit(per_page)
+    cursor = db_responses.find(search_criteria).skip(per_page * (page - 1)).limit(per_page)
     for document in cursor:
         document['_id'] = str(document['_id'])
         document['added_ms'] = int(document['added_date'].strftime("%s")) * 1000
         responses.append(document)
+
     results['results'] = responses
     return jsonify(results)
 
 
 @app.route('/responses/<mongo_id>', methods=['GET'])
 def do_get_response(mongo_id):
     try:
-        result = db.responses.find_one({"_id": ObjectId(mongo_id)})
-
+        result = get_db_responses().find_one({"_id": ObjectId(mongo_id)})
         if result:
             result['_id'] = str(result['_id'])
             return jsonify(result)
+
     except InvalidId as e:
-        return jsonify({}), 404
+        return client_error(repr(e))
+
     except Exception as e:
-        raise e
+        return server_error(e)
```

Comment:
This needs to be `return server_error(repr(e))` as otherwise Flask will try and put the exception in the error JSON and you get a `TypeError: ServerSelectionTimeoutError('mongodb:27017: [Errno -2] Name or service not known',) is not JSON serializable // Werkzeug Debugger` type error. To reproduce this try and do a GET for a document while MongoDB is not running.